### ISSUE TITLE
Mine this paper for insights

### ISSUE THREAD

**Issue by JohnGiorgi** [JohnGiorgi/t2t#10]
[This paper](https://arxiv.org/abs/1905.05583) recently performed a bunch of empirical experiments on fine-tuning BERT. There are some highly relevant insights for us:

* Rather than truncate a document with > 512 tokens, they select the first 128 and the last 382 tokens. This performs better in text classification tasks. The motivation is that these tokens _typically_ contain the most information.
* Layer-wise learning rate: they select a single learning rate and multiply it by a "decay factor", such that the bottom layers of BERT are fine-tuned with smaller effective learning rates than the top layers. This is inspired by transfer learning in computer vision.

Low priority for now, but we should investigate how these tricks affect the quality of document embeddings.

**Comment by JohnGiorgi** [comment]
Not sure we should focus on too many of these little tricks in this work. Need to stick to the core idea: contrastive learning for textual representations at the sentence / document level.
